### DISCUSSION TITLE
fuzzy import?

### DISCUSSION THREAD

**Discussion by hmijail** [windingwind/zotero-better-notes#833] [Q&A] [answered]
The ZNB readme mentions API and GUI functionality to export annotations, and API functionality to import them, but I see no GUI for importing. Is there a reason for this? Is that functionality somehow less mature?

Asking because currently Zotero has no way to migrate the annotations in a PDF to a new version of that PDF. The [official advice](https://forums.zotero.org/discussion/comment/445307#Comment_445307) to deal with this situation is to change the PDF underneath Zotero, but then the annotations might be out of place and there is no way to fix them.

So if ZBN's export and import functionality can rebuild the original annotations in an updated PDF, then I could try working on adding some kind of fuzzy matching that would try to adapt the original annotations to the new location of the text in the updated PDF. (And/or propose a Feature Request, of course :))

Does that make sense?

**Comment by windingwind** [comment, marked as answer]
If you mean import Markdown, we do (but in beta currently):

<img width="284" alt="image" src="https://github.com/windingwind/zotero-better-notes/assets/33902321/5bd88da2-bf1d-4f96-a6de-02a13df69531">

> currently Zotero has no way to migrate the annotations in a PDF to a new version of that PDF

I don't think that's possible, especially since you never know what changes there will be in two different PDFs. Short text annotations of 1 or 2 words will have multiple matches in the new version of PDF.

But, if you mean ink/text annotation, I guess that's technically doable, but definitely should go to a seperate plugin.
